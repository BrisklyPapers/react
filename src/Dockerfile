FROM node

COPY entrypoint.sh .

ENV DEBIAN_FRONTEND noninteractive
RUN apt-get update && apt-get install bash