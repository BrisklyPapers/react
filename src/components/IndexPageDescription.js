import React from 'react';
import IndexPage from '../containers/IndexPage';
import Radium from 'radium';

class IndexPageDescription extends React.Component {
    render() {
        return (
            <IndexPage>
                <div style={styles.promotion}>
                    <h2 style={styles.h2}>Document Management System</h2>
                    <p style={styles.p}>
                        Briskly Papers is a document management system. It's goal is to provide an easy and fast access to all your files.
                    </p>
                </div>
                <div style={styles.promotion}>
                    <h2 style={styles.h2}>Store your documents per Drag & Drop</h2>
                    <p style={styles.p}>
                        Select your file, move them to our Upload Area, click upload. Done
                    </p>
                </div>
                <div style={styles.promotion}>
                    <h2 style={styles.h2}>Tag your documents</h2>
                    <p style={styles.p}>
                        Create tags to define groups for your documents. You can find all relevant documents with one
                        click.
                    </p>
                </div>
                <div style={styles.promotion}>
                    <h2 style={styles.h2}>Save websites for offline usage</h2>
                    <p style={styles.p}>
                        You want to remember a great website? Enter the URL, click save and find the information
                        whenever you need it. Even offline, with our PDF creator!
                    </p>
                </div>
                <div style={styles.promotion}>
                    <h2 style={styles.h2}>Your documents are secure</h2>
                    <p style={styles.p}>
                        All documents are stored encrypted. The files can only be decrypted with your password.
                    </p>
                </div>
                <div style={styles.promotion}>
                    <h2 style={styles.h2}>Full text search</h2>
                    <p style={styles.p}>
                        Each document can be parsed and indexed. You are able to find relevant documents by tag, file name or file content
                    </p>
                </div>
            </IndexPage>
        )
    };
}

var styles = {
    h2: {
        fontSize: "34px",
        fontFamily: "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
        fontWeight: 100,
        letterSpacing: "-1px",
        margin: 0
    },
    p: {
        fontSize: "17px",
        fontFamily: "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
        letterSpacing: "0.06vmax",
        margin: "2.6vh 0 6vh",
    },
    promotion: {
        width: "40vw",
        color: "rgba(0, 0, 0, 0.4)",
        margin: "75px"
    }
};

export default Radium(IndexPageDescription);