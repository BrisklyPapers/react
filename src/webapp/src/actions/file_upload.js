import $ from 'jquery';

export const FILE_DROP = 'FILE_DROP';
export const fileDropped = (document) => {
    return {
        type: FILE_DROP,
        document
    }
};

export const FILE_BUFFER_FILE = 'FILE_BUFFER_FILE';
export const fileBufferFile = (document) => {
    return {
        type: FILE_BUFFER_FILE,
        document
    }
};

export const FILE_UPLOAD_CLICKED = 'FILE_UPLOAD_CLICKED';
export const fileUploadClicked = () => {
    return {
        type: FILE_UPLOAD_CLICKED
    }
};

export const FILE_BOX_CHANGED = 'FILE_BOX_CHANGED';
export const fileBoxChanged = (documents) => {
    return {
        type: FILE_BOX_CHANGED,
        documents
    }
};

export const FILE_START_UPLOAD = 'FILE_START_UPLOAD';
export const fileStartUpload = (documents) => {
    return {
        type: FILE_START_UPLOAD,
        documents
    }
};

export const FILE_UPLOAD_FINISHED = 'FILE_UPLOAD_FINISHED';
export const fileUploadFinished = (documents) => {
    return {
        type: FILE_UPLOAD_FINISHED,
        documents
    }
};


export const storeDocuments = (files, tags) => {

    return function (dispatch) {
        var formData = new FormData();

        $.each(files, function (i, file) {
            formData.append('files[]', file);
        });

        $.each(tags, function (i, tag) {
            formData.append("tags[" + tag.key + "]", tag.label);
        });

        formData.append('action', 'upload');

        return $.ajax({
            url: '  http://localhost:8085/document',
            data: formData,
            type: "post",
            contentType: false,
            processData: false,
            success: function (jqXHR, textStatus) {
                dispatch(documentsStored({}));
            },
            error: function (jqXHR, textStatus) {
                dispatch(documentsNotStored({}));
            }
        })
    }
};

export const DOCUMENTS_STORED = 'DOCUMENTS_STORED';
const documentsStored = (json) => {
    return {
        type: DOCUMENTS_STORED,
        documents: json.map ? json : [],
        receivedAt: Date.now()
    }
};

export const DOCUMENTS_NOT_STORED = 'DOCUMENTS_NOT_STORED';
const documentsNotStored = () => {
    return {
        type: DOCUMENTS_NOT_STORED,
        documents: [],
        receivedAt: Date.now()
    }
};
