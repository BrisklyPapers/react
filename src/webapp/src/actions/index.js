import fetch from 'isomorphic-fetch'

export const SEARCH_INPUT_KEY_PRESSED = 'SEARCH_INPUT_KEY_PRESSED';
export function searchInputKeyPressed(text) {
    return {
        type: SEARCH_INPUT_KEY_PRESSED,
        text
    }
}

export const SEARCH_DOCUMENTS = 'SEARCH_DOCUMENTS';
export function searchDocuments(text) {
    return {
        type: SEARCH_DOCUMENTS,
        text
    }
}

export const RECEIVE_DOCUMENTS = 'RECEIVE_DOCUMENTS';
function receiveDocuments(json) {
    console.log(json.data);
    return {
        type: RECEIVE_DOCUMENTS,
        documents: [], //json.data.children.map(child => child.data),
        receivedAt: Date.now()
    }
}

export const RECEIVE_DOCUMENTS_ERROR = 'RECEIVE_DOCUMENTS_ERROR';
function receiveDocumentsError(json) {
    return {
        type: RECEIVE_DOCUMENTS_ERROR,
        documents: [],
        receivedAt: Date.now()
    }
}

// store.dispatch(fetchPosts())
export function fetchDocuments(text) {
    // Thunk middleware knows how to handle functions.
    // It passes the dispatch method as an argument to the function,
    // thus making it able to dispatch actions itself.

    return function (dispatch) {
        // First dispatch: the app state is updated to inform
        // that the API call is starting.

        dispatch(searchDocuments(text))

        // The function called by the thunk middleware can return a value,
        // that is passed on as the return value of the dispatch method.

        // In this case, we return a promise to wait for.
        // This is not required by thunk middleware, but it is convenient for us.

        return fetch(`http://localhost:8085/document/search?q=${text}`)
            .then(
                response => response.json(),
                // Do not use catch, because that will also catch
                // any errors in the dispatch and resulting render,
                // causing an loop of 'Unexpected batch number' errors.
                // https://github.com/facebook/react/issues/6895
                error => console.log('An error occured.', error)
            )
            .then(json =>
                // We can dispatch many times!
                // Here, we update the app state with the results of the API call.

                dispatch(receiveDocuments(json))
            )
    }
}

export function fetchDocumentsIfNeeded(text) {
    // Note that the function also receives getState()
    // which lets you choose what to dispatch next.

    // This is useful for avoiding a network request if
    // a cached value is already available.

    return (dispatch, getState) => {
        if (1) {
            // Dispatch a thunk from thunk!
            return dispatch(fetchDocuments(text))
        } else {
            // Let the calling code know there's nothing to wait for.
            return Promise.resolve()
        }
    }
}