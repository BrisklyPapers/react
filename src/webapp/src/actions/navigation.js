export const PAGEVIEW_SEARCH_RESULTS = 'PAGEVIEW_SEARCH_RESULTS';
export const showResultPage = () => {
    return {
        type: PAGEVIEW_SEARCH_RESULTS,
    }
};

export const PAGEVIEW_FILE_UPLOAD = 'PAGEVIEW_FILE_UPLOAD';
export const showFileUploadPage = () => {
    return {
        type: PAGEVIEW_FILE_UPLOAD,
    }
};
