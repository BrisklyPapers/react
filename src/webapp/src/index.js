import 'babel-polyfill'

import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import thunkMiddleware from 'redux-thunk'
import { createLogger } from 'redux-logger'
import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from './reducers';
import App from './components/App';

// Needed for onTouchTap
// http://stackoverflow.com/a/34015469/988941
injectTapEventPlugin();

const loggerMiddleware = createLogger();

let preloadedState = {searchResults: {documents: []}};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/* eslint-disable no-underscore-dangle */
let store = createStore(
    rootReducer,
    undefined,
    composeEnhancers(applyMiddleware(
        thunkMiddleware, // lets us dispatch() functions
        loggerMiddleware // neat middleware that logs actions
    ))
);
/* eslint-enable */

render(
    <MuiThemeProvider>
        <Provider store={store}>
            <App />
        </Provider>
    </MuiThemeProvider>,
    document.getElementById('root')
);