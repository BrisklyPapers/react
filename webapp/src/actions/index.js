export * from './search';
export * from './file_upload';
export * from './navigation';