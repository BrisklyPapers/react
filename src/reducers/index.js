import {combineReducers} from 'redux';
import {searchedText} from './searchedText';
import {searchResults} from './searchResults';
import {droppedFiles} from './droppedFiles';
import {uploading} from './uploading';
import {upload_error} from './upload_error';
import {navigation} from './navigation';
import {logged_in} from './logged_in';

export {combineReducers} from 'redux';
export {searchedText} from './searchedText';
export {searchResults} from './searchResults';
export {droppedFiles} from './droppedFiles';
export {uploading} from './uploading';
export {upload_error} from './upload_error';
export {navigation} from './navigation';
export {logged_in} from './logged_in';

const rootReducer = combineReducers({
    searchedText,
    searchResults,
    droppedFiles,
    uploading,
    upload_error,
    navigation,
    logged_in
});

export default rootReducer
