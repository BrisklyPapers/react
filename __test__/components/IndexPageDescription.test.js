import React from 'react';
import IndexPageDescription from '../../src/components/IndexPageDescription.js';
import {PAGEVIEW_INDEX} from '../../src/actions';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {spy} from 'sinon';
import fetchMock from 'fetch-mock'
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureMockStore();

describe('components/IndexPageDescription', () => {
    it('renders correctly', () => {
        const search = renderer.create(
            <Provider store={mockStore({navigation: PAGEVIEW_INDEX})}>
                <IndexPageDescription />
            </Provider>
        );
        expect(search.toJSON()).toMatchSnapshot();
    });
});