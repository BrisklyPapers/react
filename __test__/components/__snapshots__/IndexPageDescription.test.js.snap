// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IndexPageDescription renders correctly 1`] = `
<div>
  <div
    data-radium={true}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.4)",
        "margin": "75px",
        "width": "40vw",
      }
    }
  >
    <h2
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "34px",
          "fontWeight": 100,
          "letterSpacing": "-1px",
          "margin": 0,
        }
      }
    >
      Document Management System
    </h2>
    <p
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "17px",
          "letterSpacing": "0.06vmax",
          "margin": "2.6vh 0 6vh",
        }
      }
    >
      Briskly Papers is a document management system. It's goal is to provide an easy and fast access to all your files.
    </p>
  </div>
  <div
    data-radium={true}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.4)",
        "margin": "75px",
        "width": "40vw",
      }
    }
  >
    <h2
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "34px",
          "fontWeight": 100,
          "letterSpacing": "-1px",
          "margin": 0,
        }
      }
    >
      Store your documents per Drag & Drop
    </h2>
    <p
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "17px",
          "letterSpacing": "0.06vmax",
          "margin": "2.6vh 0 6vh",
        }
      }
    >
      Select your file, move them to our Upload Area, click upload. Done
    </p>
  </div>
  <div
    data-radium={true}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.4)",
        "margin": "75px",
        "width": "40vw",
      }
    }
  >
    <h2
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "34px",
          "fontWeight": 100,
          "letterSpacing": "-1px",
          "margin": 0,
        }
      }
    >
      Tag your documents
    </h2>
    <p
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "17px",
          "letterSpacing": "0.06vmax",
          "margin": "2.6vh 0 6vh",
        }
      }
    >
      Create tags to define groups for your documents. You can find all relevant documents with one click.
    </p>
  </div>
  <div
    data-radium={true}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.4)",
        "margin": "75px",
        "width": "40vw",
      }
    }
  >
    <h2
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "34px",
          "fontWeight": 100,
          "letterSpacing": "-1px",
          "margin": 0,
        }
      }
    >
      Save websites for offline usage
    </h2>
    <p
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "17px",
          "letterSpacing": "0.06vmax",
          "margin": "2.6vh 0 6vh",
        }
      }
    >
      You want to remember a great website? Enter the URL, click save and find the information whenever you need it. Even offline, with our PDF creator!
    </p>
  </div>
  <div
    data-radium={true}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.4)",
        "margin": "75px",
        "width": "40vw",
      }
    }
  >
    <h2
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "34px",
          "fontWeight": 100,
          "letterSpacing": "-1px",
          "margin": 0,
        }
      }
    >
      Your documents are secure
    </h2>
    <p
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "17px",
          "letterSpacing": "0.06vmax",
          "margin": "2.6vh 0 6vh",
        }
      }
    >
      All documents are stored encrypted. The files can only be decrypted with your password.
    </p>
  </div>
  <div
    data-radium={true}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.4)",
        "margin": "75px",
        "width": "40vw",
      }
    }
  >
    <h2
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "34px",
          "fontWeight": 100,
          "letterSpacing": "-1px",
          "margin": 0,
        }
      }
    >
      Full text search
    </h2>
    <p
      data-radium={true}
      style={
        Object {
          "fontFamily": "HelveticaNeueW01-45Ligh,HelveticaNeueW02-45Ligh,HelveticaNeueW10-45Ligh,sans-serif",
          "fontSize": "17px",
          "letterSpacing": "0.06vmax",
          "margin": "2.6vh 0 6vh",
        }
      }
    >
      Each document can be parsed and indexed. You are able to find relevant documents by tag, file name or file content
    </p>
  </div>
</div>
`;
